### PR TITLE
Connect game character with user

### PR DESCRIPTION
**Pull request by przemilymaciej**

### CODE REVIEW

File: src/models/GameCharacter.ts [hunk 1]
```diff
@@ -22,6 +27,7 @@ const GameCharacterSchema = new mongoose.Schema({
   skills: { type: Array },
   numberOfFights: { type: Number, required: true, default: 0 },
   wins: { type: Number, required: true, default: 0 }
```
Comment (bartlomiejmont) [inline src/models/GameCharacter.ts]: , is missing

File: src/models/GameCharacter.ts [hunk 2]
```diff
@@ -4,9 +4,14 @@ export interface IGameCharacter extends mongoose.Document{
...
-    owner: string,
     power: number,
     attackSpeed: number,
+    ConnectGameCharacterWithUser
+    skills: Array,
+    User: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "User"
+    }
     skills: Array<String>,
```
Comment (bartlomiejmont) [inline src/models/GameCharacter.ts:13]: Duplicated array skills

File: src/models/GameCharacter.ts [hunk 3]
```diff
@@ -4,9 +4,14 @@ export interface IGameCharacter extends mongoose.Document{
     name: string,
     level: number,
     healthLevel: number,
-    owner: string,
     power: number,
     attackSpeed: number,
+    ConnectGameCharacterWithUser
```
Comment (bartlomiejmont) [inline src/models/GameCharacter.ts]: what is this ?

File: src/models/GameCharacter.ts [hunk 4]
```diff
@@ -4,9 +4,14 @@ export interface IGameCharacter extends mongoose.Document{
...
+    skills: Array,
+    User: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "User"
+    }
```
Comment (bartlomiejmont) [inline src/models/GameCharacter.ts]: Interface should have `userId: string` field instead of User object reference

**Review by bartlomiejmont — CHANGES_REQUESTED** [review]
All in all your connection is fine but I think  relation User->GameCharacter will fit our needs better than GameCharacter->User

File: src/models/GameCharacter.ts (hunk 1 above)
Comment (przemilymaciej) [reply src/models/GameCharacter.ts]: done

File: src/models/GameCharacter.ts (hunk 2 above)
Comment (przemilymaciej) [reply src/models/GameCharacter.ts:13]: done

File: src/models/GameCharacter.ts (hunk 4 above)
Comment (przemilymaciej) [reply src/models/GameCharacter.ts]: done

**Review by bartlomiejmont — APPROVED** [review]

**Pull request merged**